pkg/handler/processor/guesser/type_dsse.go

```diff
@@ -0,0 +1,41 @@
+//
+// Copyright 2022 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package guesser
+
+import (
+	"encoding/json"
+
+	"github.com/guacsec/guac/pkg/handler/processor"
+	"github.com/secure-systems-lab/go-securesystemslib/dsse"
+)
+
+type dsseTypeGuesser struct{}
+
+type envelope struct {
```

Comment:
can we use https://github.com/secure-systems-lab/go-securesystemslib/blob/main/dsse/sign.go#L28?

Reply:
@lumjjb @pxp928 if we use the `dsse.Envelope` struct, we would have to assert that the fields in the struct are not empty strings. Would we want the envelope below to be marked as `DocumentUnknown`? If so, I'll make the switch
```json
{
  "payload": "",
  "payloadType": "",
  "signatures": [...]
}
```

Reply:
Yes that is true. Otherwise it would be a malformed DSSE envelope.